ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CommonDataTypeVerifier.kt

```diff
@@ -20,177 +20,78 @@ import org.codice.compliance.SAMLCore_1_3_2_a
 import org.w3c.dom.Node
 import java.net.URI
-import java.time.format.DateTimeFormatter
-import java.time.format.DateTimeParseException
 
 class CommonDataTypeVerifier {
     companion object {
-        private const val FOUR_DIGIT_YEAR_LEN = 4
-        private const val NO_HYPHEN_MESSAGE = "No hyphen was found in the date value of"
-
         var ids = mutableListOf<String>()
 
         /** 1.3 Common Data Types **/
         fun verifyCommonDataType(samlDom: Node) {
-            ids = mutableListOf()
-
             for (i in samlDom.childNodes.length - 1 downTo 0) {
                 val child = samlDom.childNodes.item(i)
-                val typeAttribute = child.attributes?.getNamedItemNS(XSI, "type")
-                if (typeAttribute?.textContent?.contains("string") == true)
-                    verifyStringValues(child, null)
-                if (typeAttribute?.textContent?.contains("anyURI") == true)
-                    verifyUriValues(child, null)
-                if (typeAttribute?.textContent?.contains("dateTime") == true)
-                    verifyTimeValues(child, null)
-                if (typeAttribute?.textContent?.contains("ID") == true)
-                    verifyIdValues(child, null)
+                val type = child.attributes.getNamedItemNS(XSI, "type").textContent
+                when {
+                    type.contains("string") -> verifyStringValues(child)
+                    type.contains("anyURI") -> verifyUriValues(child)
+                    type.contains("dateTime") -> verifyDateTimeValues(child)
+                    type.contains("ID") -> verifyIdValues(child)
+                }
 
                 if (child.hasChildNodes())
                     verifyCommonDataType(child)
             }
         }
 
         /** 1.3.1 String Values **/
-        @Suppress("StringLiteralDuplication")
-        fun verifyStringValues(node: Node, errorCode: SAMLSpecRefMessage?) {
+        fun verifyStringValues(node: Node, errorCode: SAMLSpecRefMessage? = null) {
             if (StringUtils.isBlank(node.textContent)) {
+                val errorMessage = "The String value of ${node.textContent} is invalid."
                 if (errorCode != null) throw SAMLComplianceException.create(errorCode,
                         SAMLCore_1_3_1_a,
-                        message = "The String value of ${node.textContent} is invalid.")
+                        message = errorMessage)
                 else throw SAMLComplianceException.create(SAMLCore_1_3_1_a,
-                        message = "The String value of ${node.textContent} is invalid.")
+                        message = errorMessage)
             }
         }
 
         /** 1.3.2 URI Values **/
-        fun verifyUriValues(node: Node, errorCode: SAMLSpecRefMessage?) {
+        fun verifyUriValues(node: Node, errorCode: SAMLSpecRefMessage? = null) {
             if (StringUtils.isBlank(node.textContent)
                     && !URI.create(node.textContent).isAbsolute) {
```

Comment:
❗️  Was just looking at this again because of something else I'm doing. Shouldn't this be an `OR` operator not an `AND` operator? Those two conditions are never going to be true at the same time.